src/librustc_typeck/diagnostics.rs

```diff
@@ -4095,6 +4095,27 @@ assert_eq!(!Question::No, true);
 ```
 "##,
 
+E0608: r##"
+An attempt to get an index from a type which doesn't implement the `Index` was
```

Comment:
This isn't really getting an index, at least in the eyes of the user, is it? It's more trying to index into the type. Maybe rephrase this as "An attempt to index into a type which doesn't implement the `Index` trait was performed." Maybe it actually makes sense to use `std::ops::Index` instead of `Index` as well.